### PR TITLE
Use is_x86_feature_detected instead of CPUID if std is available

### PR DESCRIPTION
**Pull request by akash-fortanix**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -10,3 +10,7 @@ documentation = "https://docs.rs/rdrand/0.3.0/"
 
 [dependencies]
 rand_core = { version = "0.3", default-features = false }
+
+[features]
+default = ["std"]
+std = []
```
Comment (jethrogb) [inline Cargo.toml]: No newline at end of file

File: src/lib.rs [hunk 2]
```diff
@@ -171,13 +185,11 @@ macro_rules! impl_rand {
...
             /// supported, an error is returned.
             pub fn new() -> Result<Self, Error> {
-                unsafe {
-                    if cfg!(target_feature=$feat) || check_cpuid!($feat) {
-                        Ok($gen(()))
-                    } else {
-                        Err(Error::new(rand_core::ErrorKind::Unavailable,
-                                       "the instruction is not supported"))
-                    }
+                if cfg!(target_feature=$feat) || is_feature_supported!($feat) {
```
Comment (jethrogb) [inline src/lib.rs]: `is_x86_feature_detected` already checks `cfg!(target_feature`

File: src/lib.rs [hunk 3]
```diff
@@ -146,6 +150,16 @@ macro_rules! check_cpuid {
     } };
 }
 
+#[cfg(not(feature = "std"))]
+macro_rules! is_feature_supported {
```
Comment (jethrogb) [inline src/lib.rs]: call this macro `is_x86_feature_detected`, then you no longer need the `#[cfg(feature = "std")] macro_rules! is_feature_supported` case

**Review by jethrogb — CHANGES_REQUESTED** [review]

**Comment by akash-fortanix** [comment]
filed PR against original repo, the purpose of this PR is served

**Pull request closed without merging**